ERA/ERi.nls

```diff
@@ -1,43 +1,96 @@
-      set index-counter (index-counter + 1)
+
+  let curr-currency 0
+  while [curr-currency < length CURRENCY-VALUES] [
+    set CURRENCY-VALUES replace-item curr-currency CURRENCY-VALUES [ERiC] of one-of ops-nodes with [who = curr-currency] ;;theres only one, ignore the one-of used to grab the only ops-node
+    set curr-currency curr-currency + 1
+  ]
+  let base-exchange-rates (list)
+  let currency-row 0
+  let exch-rate table:make
+  while [currency-row < num-jurisdictions + num-decentralized-currencies] [
+    let currency-column 0
+    while [currency-column < num-jurisdictions + num-decentralized-currencies] [
+      let exchange item currency-column CURRENCY-VALUES / item currency-row CURRENCY-VALUES
+      table:put exch-rate currency-column exchange
+      set currency-column currency-column + 1
+    ]
+    set currency-row currency-row + 1
+    set base-exchange-rates lput (copy-table exch-rate) base-exchange-rates
+  ]
+  set BASE-YEAR-EXCHANGE-RATES base-exchange-rates
+end
+
+to ERi.recalculate-ERiE ;called by observer
+  let new-ERiE table:make
+  let soil-health-table table:make
+  let eco 0
+  let curr-soil-health 0
+  let curr-eco "A"
+  while [eco < num-ecoregions] [
+    set curr-eco (index-to-ecoregion eco)
+    ifelse count patches with [ecoregion = curr-eco] = 0 [
+      set curr-soil-health 0
+    ]  [
+      set curr-soil-health min (list 1 ((mean [soil-health] of patches with [ecoregion = curr-eco] - max-soil-health / 2) / max-soil-health / 2)) ;;normalizes worst case to -1 and best case to 1
     ]
+    table:put soil-health-table curr-eco curr-soil-health
+    table:put new-ERiE curr-eco (Bezier (curr-soil-health) 0 1)
+    set eco eco + 1
+  ]
+  set PREVIOUS-SOIL-HEALTH soil-health-table
+  set ERiE new-ERiE
+end
+
+to ERi.recalculate-ERiC ;called by ops-nodes
+  let eco 0
+  let eric-temp 0
+  while [eco < num-ecoregions] [
+    set eric-temp eric-temp + ((table:get ERiE (index-to-ecoregion eco)) * (ecoregion-percent (index-to-ecoregion eco) who))
+    set eco eco + 1
   ]
-;  show CURRENT-EXCHANGE-RATES
-;  set BASE-YEAR-EXCHANGE-RATES original-base-year-exchange-rates
+  if node-jurisdiction != "decentralized" [ set ERiC eric-temp ]
+end
+
+to-report ecoregion-percent [eco jur]
+  if count patches with [jurisdiction = jur] = 0 [report 0] ;; this is a catch all for decentralized op-node and op-nodes with no patches
+  report count patches with [Ecoregion = eco and jurisdiction = jur] / count patches with [jurisdiction = jur]
 end
 
+to-report Bezier [unadjusted-erie lower-bound upper-bound] ;https://javascript.info/bezier-curve inputs the soil health and outputs the 
+  ;numbers currently hardcoded but we can change that later with sliders
+  if unadjusted-erie >= 1 [report 0.3]
+  if unadjusted-erie <= -1 [report 10]
+  
+  let t ((upper-bound - lower-bound) / 2) + lower-bound
+  (ifelse precision ((1 - t) ^ 3 + 0.2 * (1 - t) ^ 2 * t + -3 * (1 - t) * t ^ 2 + -1 * t ^ 3) 2 = precision unadjusted-erie 2 [
```

Comment:
Abstract `((1 - t) ^ 3 + 0.2 * (1 - t) ^ 2 * t + -3 * (1 - t) * t ^ 2 + -1 * t ^ 3)` into a reporter and call that below instead of repeating this complicated formulat.